### ISSUE TITLE
Unable to use `eslint-disable` comment to disable a rule that reports `loc` with a violation

### ISSUE THREAD

**Issue by bmish** [eslint/eslint#12245]
**Tell us about your environment**

* **ESLint Version:** 5.16.0, 6.3.0
* **Node Version:** 8.15.1, 10.16.3
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parser: 'babel-eslint',
  rules: {
    'my-custom-rule': 'error'
  },
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// my-custom-rule.js
module.exports = {
  create(context) {
    return {
      Program(node) {
        context.report({
          node,
          loc: {
            start: { line: 0, column: 1 },
            end : { line: 1, column: 2 }
          },
          message: 'Some error message'
        });
      }
    };
  }
};
```

```js
// my-file.js
/* eslint-disable my-custom-rule */
var x = 123;
console.log(x);
```

```bash
yarn run eslint my-file.js
```

**What did you expect to happen?**

`my-custom-rule` should be disabled because of the `eslint-disable` comment and should not report a violation in my-file.js.

**What actually happened? Please include the actual, raw output from ESLint.**

`my-custom-rule` reported a violation in my-file.js even though it should have been disabled for the file.

```
my-file.js
  0:2  error  Some error message  my-custom-rule
```

Removing `loc` from `context.report` results in the correct behavior. Passing `loc` to `context.report` appears to cause the rule to ignore the `eslint-disable` comment at the top of the file.

**Are you willing to submit a pull request to fix this bug?**
Possibly.

**Comment by mysticatea** [comment]
Thank you for this issue.

Because `/*eslint-disable*/` comment doesn't disable rules on the entire file. It disables rules between the end of a `/*eslint-disable*/` comment and the start of the next `/*eslint-enable*/` comment. In your example case, it disables `my-custom-rule` after `1:36`.

**Comment by platinumazure** [comment]
In order to disable a rule on the whole file, you should use `/* eslint rule-name: "off" */`.

The `/* eslint-disable */` comments disable from the end of the comment. I could see an argument to allow them to disable from the start of the comment, but that would need to be evaluated as a core enhancement (and would also be a breaking change).

**Comment by bmish** [comment]
Thank you for the explanations! The current behavior makes sense now. And switching to using `/* eslint rule-name: "off" */` achieves exactly what I want here.
